### ISSUE TITLE
dbx sync repo should not have default profile

### ISSUE THREAD

**Issue by ep-mo** [databrickslabs/dbx#833]
## Expected Behavior
`dbx sync repo --dest-repo=test` should default to use `$DATABRICKS_HOST` and `$DATABRICKS_TOKEN` environment variables for authentication.

## Current Behavior
`dbx sync repo --dest-repo=test` defaults to `--profile=DEFAULT`, because argument `--profile` is specified with `[default: DEFAULT]`. This means that the current default behavior is to use config file instead of environment variables (as recommended for CI/CD pipelines). Since our CI/CD pipeline does not have a config file, we get the following error: `Could not find a databricks-cli config for profile DEFAULT`

Our current workaround is to set an empty profile:
`dbx sync repo --dest-repo=test --profile=`

## Your Environment

* dbx version used: 0.8.18
* Databricks Runtime version:

**Comment by CristianSmau** [comment]
I have the same issue.

However, as a workaround for this, the databrikcs-cli can be set on the DEFAULT profile by running an additional step as follows:

  - bash: |
      DATABRICKS_HOST='https://adb-XXXXXXXXXXX.azuredatabricks.net/'
      DATABRICKS_TOKEN='XXXXXXXXXXXXXXXXXXXXXXXXXXX'
      databricks configure --token --profile DEFAULT <<EOF
      $DATABRICKS_HOST
      $DATABRICKS_TOKEN
      EOF
    displayName: 'Databricks Configuration for repo Sync'

**Comment by ep-mo** [comment]
The workaround we use in our CI/CD pipeline does not require a cfg file, and is quite simple. We just pass an empty `--profile=` argument (as described in original issue), and dbx will fallback to using `$DATABRICKS_HOST` and `$DATABRICKS_TOKEN`. It took a process of trial and error to figure out..

> Our current workaround is to set an empty profile:
> dbx sync repo --dest-repo=test **--profile=**

**Comment by ep-mo** [comment]
> does this mean the empty profile workaround only works if the file is not present?

No, I don't think it matters, the workaround should work even if the cfg file is present (from the top of my head). When you specify `--profile=`, dbx should use your environment variables.

In the **Steps to Reproduce** section I just tried to explain the steps to reproduce the issue and how to get a stack trace. If you have a cfg file when you try to reproduce, it would just fallback to your DEFAULT profile in your cfg file, and everything works fine (if you have said file).

**Comment by CristianSmau** [comment]
Looks like [DBX](https://docs.databricks.com/en/archive/dev-tools/dbx/dbx.html) got deprecated and replaced by [ADB ](https://docs.databricks.com/en/dev-tools/bundles/index.html). Worth looking into the new package.

**Comment by ep-mo** [comment]
> Looks like [DBX](https://docs.databricks.com/en/archive/dev-tools/dbx/dbx.html) got deprecated and replaced by [ADB ](https://docs.databricks.com/en/dev-tools/bundles/index.html). Worth looking into the new package.

From what I understand, dbx is not yet deprecated and is still maintained by Databricks Labs ([source](https://github.com/databrickslabs/dbx/issues/806#issuecomment-1621755806)). Our CI/CD is built around dbx, so we will continue to use dbx, probably at least until DAB is generally available. But yeah, worth looking into the new package. If we started from scratch today, we would probably look into DAB, because it's expected to supersede dbx.
